### PR TITLE
Add -iter option to pkcs12 command

### LINKED ISSUE #8194: Set number of iterations for `openssl pkcs12` command
The `openssl pkcs12` command doesn't have options to set the number of key iterations and MAC iterations. It uses 2048 by default and they can be set to 1 by using `-noiter` and `-nomaciter`.

I'm proposing to add two new parameters to allow the user to set an arbitrary number of iterations. Unfortunately, `-maciter` is already taken by a legacy option, so I suggest calling these new options `-iters n` and `-maciters n`.

### PR DESCRIPTION
**Pull request by imgx64**
Fixes #8194

##### Checklist

- [x] documentation is added or updated
- [ ] tests are added or updated _(Not sure how to add a test for a CLI command)_

### CODE REVIEW

**Comment by richsalz** [comment]
Please wait until #9953 is done before merging this.

File: doc/man1/openssl-pkcs12.pod [hunk 1]
```diff
@@ -251,23 +251,33 @@ the use of signing only keys for SSL client authentication.
...
+The MAC is used to check the file integrity but since it will normally
 have the same password as the keys and certificates it could also be attacked.
-By default both MAC and encryption iteration counts are set to 2048, using
-these options the MAC and encryption iteration counts can be set to 1, since
-this reduces the file security you should not use these options unless you
-really have to. Most software supports both MAC and key iteration counts.
-MSIE 4.0 doesn't support MAC iteration counts so it needs the B<-nomaciter>
-option.
+
+By default MAC iteration count is set to 2048, Using this option sets it to 1.
```
Comment (mattcaswell) [inline doc/man1/openssl-pkcs12.pod]: Typo: *the* MAC iteration count

File: apps/pkcs12.c [hunk 2]
```diff
@@ -84,7 +84,8 @@ const OPTIONS pkcs12_options[] = {
     {"certpbe", OPT_CERTPBE, 's', "Certificate PBE algorithm (default 3DES)"},
 # endif
     {"export", OPT_EXPORT, '-', "Output PKCS12 file"},
-    {"noiter", OPT_NOITER, '-', "Don't use encryption iteration"},
+    {"iter", OPT_ITER, 'p', "Specify the encryption key iteration count"},
+    {"noiter", OPT_NOITER, '-', "Don't use encryption key iteration"},
```
Comment (mattcaswell) [inline apps/pkcs12.c]: Should we disallow specifying both "-iter" and "-noiter" in combination? It makes no sense to allow it.

**Comment by mattcaswell** [comment]
Close/reopen to kick CLA bot.

File: doc/man1/openssl-pkcs12.pod (hunk 1 above)
Comment (imgx64) [reply doc/man1/openssl-pkcs12.pod]: Fixed.

File: apps/pkcs12.c (hunk 2 above)
Comment (imgx64) [reply apps/pkcs12.c]: It's already possible to specify multiple `-iter` options (in `pkcs8` command for example), and the last one takes efffect. `-noiter` is simply an alias for `-iter 1` so I think it's fine to allow both.

**Comment by mattcaswell** [comment]
Needs a second reviewer.

File: doc/man1/openssl-pkcs12.pod.in [hunk 3]
```diff
@@ -237,23 +237,33 @@ the use of signing only keys for SSL client authentication.
...
 
 To discourage attacks by using large dictionaries of common passwords the
 algorithm that derives keys from passwords can have an iteration count applied
 to it: this causes a certain part of the algorithm to be repeated and slows it
-down. The MAC is used to check the file integrity but since it will normally
+down. The default value is 2048.
+
+=item B<-noiter>
+
+Don't use encryption key iteration. Equivalent to B<-iter 1>.
```
Comment (richsalz) [inline doc/man1/openssl-pkcs12.pod.in]: Ping @t8m or @levitte should that ```B<>``` be ```C<>```?  Or maybe just say "an interation of one" and avoid the whole mess.

Comment (t8m) [reply doc/man1/openssl-pkcs12.pod.in]: I think it should be `C<-iter 1>`

**Comment by t8m** [comment]
I'd suggest setting the maciter value as well when -iter is used. In general the iteration count should be the same for encryption and mac as usually the same password is used for both.

File: doc/man1/openssl-pkcs12.pod.in (hunk 3 above)
Comment (levitte) [reply doc/man1/openssl-pkcs12.pod.in]: C

**Comment by imgx64** [comment]
> consider setting maciter with the -iter option.

Good idea. I was going to add another option for maciter (In a separate PR) but I wasn't sure of the name since `-maciter` is already taken. Setting both with the same option solves this issue.

Incidentally, this makes the documentation change moot since `-noiter` won't be equivalent to `-iter 1` anymore.

**Comment by imgx64** [comment]
@t8m May I also make `-maciter` a true no-op with this change? It currently sets maciter to 2048 (the default) so it's useless except to reverse a previous `-nomaciter`. But if I make `-iter` set maciter and someone does `-iter 50000 -maciter`, they would get iter=50000 and maciter=2048 which is confusing to say the least.

**Comment by t8m** [comment]
Yep, I'd say it makes sense making it no-op, please document it. Or alternative would be to make it to assign the maciter from iter instead of using the default. This way it would still be possible to revert the -nomaciter with it.

**Comment by imgx64** [comment]
@t8m 
> Yep, I'd say it makes sense making it no-op, please document it.

It's already documented as:
```
This option is included for compatibility with previous versions, it used
to be needed to use MAC iterations counts but they are now used by default.
```
To me, it looks clear that it's a no-op. I can rephrase it though if you wish.

> Or alternative would be to make it to assign the maciter from iter instead of using the default. This way it would still be possible to revert the -nomaciter with it.

That proved to be too confusing. As `-maciter -iter $n` would become different from ` -iter $n -maciter`. I just settled for setting both with `-iter` and making `-maciter` a no-op.

File: doc/man1/openssl-pkcs12.pod.in [hunk 4]
```diff
@@ -233,17 +234,19 @@ the use of signing only keys for SSL client authentication.
 
 Specify the MAC digest algorithm. If not included them SHA1 will be used.
 
-=item B<-nomaciter>, B<-noiter>
+=item B<-iter> I<count>
 
-These options affect the iteration counts on the MAC and key algorithms.
-Unless you wish to produce files compatible with MSIE 4.0 you should leave
-these options alone.
+This option specifies the iteration count for encryption key and MAC. The
```
Comment (mattcaswell) [inline doc/man1/openssl-pkcs12.pod.in]: Nit: for *the* encryption key

**Comment by mattcaswell** [comment]
> I assume @mattcaswell 's review still holds.

It does.

**Comment by t8m** [comment]
Merged to master as 1fdde9170c.

**Pull request closed without merging**